lms/services/jstor.py

```diff
@@ -62,28 +115,46 @@ def _get_public_url(self, url):
         :raises ExternalRequestError: If we get bad data back from the service
         """
 
-        doi = url.replace("jstor://", "")
-        if "/" not in doi:
-            doi = f"{self.DEFAULT_DOI_PREFIX}/{doi}"
+        article_id = url.replace("jstor://", "")
+        doi = self._doi_from_article_id(article_id)
+        s3_url = self._api_request(format_uri("/pdf-url/{}", doi)).text
 
+        if not s3_url.startswith("https://"):
+            raise ExternalRequestError(
+                f"Expected to get an S3 URL but got: '{s3_url}' instead"
+            )
+
+        return s3_url
+
+    def _api_request(self, endpoint: str) -> requests.Response:
```

Comment:
I was going to suggest doing the `format_uri` here instead than from the caller but that will only work nicely as long as what's true about this two endpoints is true for any new ones.